hvac/api/auth_methods/aws.py

```diff
@@ -0,0 +1,767 @@
+
+        Supported methods:
+            GET: /auth/{mount_point}/config/identity. Produces: 200 application/json
+
+        :param mount_point: The "path" the aws auth method was mounted on
+        :type mount_point: str | unicode
+        :return: The data key from the JSON response of the request.
+        :rtype: dict
+        """
+        api_path = '/v1/auth/{mount_point}/config/identity'.format(mount_point=mount_point)
+        response = self._adapter.get(
+            url=api_path,
+        )
+        return response.json().get('data')
+
+    def create_certificate_configuration(self, cert_name, aws_public_cert, document_type="pkcs7", mount_point=AWS_DEFAULT_MOUNT_POINT):
+        """Registers an AWS public key to be used to verify the instance identity documents
+
+        While the PKCS#7 signature of the identity documents have DSA digest, the identity signature will have RSA
+        digest, and hence the public keys for each type varies respectively. Indicate the type of the public key using
+        the "type" parameter
+
+        Supported methods:
+            POST: /auth/{mount_point}/config/certificate/:cert_name Produces: 204 (empty body)
+
+        :param cert_name: Name of the certificate
+        :type cert_name: string | unicode
+        :param aws_public_cert: Base64 encoded AWS Public key required to verify PKCS7 signature of the EC2 instance
+            metadata
+        :param document_type: Takes the value of either "pkcs7" or "identity", indicating the type of document which can be
+            verified using the given certificate
+        :type document_type: string | unicode
+        :param mount_point: The "path" the aws auth method was mounted on
+        :type mount_point: str | unicode
+        :return: The response of the request
+        :rtype: request.Response
+        """
+        params = {
+            'cert_name': cert_name,
+            'aws_public_cert': aws_public_cert,
+            'document_type': document_type,
+        }
+        api_path = '/v1/auth/{0}/config/certificate/{1}'.format(mount_point, cert_name)
+        return self._adapter.post(
+            url=api_path,
+            json=params,
+        )
+
+    def read_certificate_configuration(self, cert_name, mount_point=AWS_DEFAULT_MOUNT_POINT):
+        """Returns the previously configured AWS public key
+
+        Supported methods:
+            GET: /v1/auth/{mount_point}/config/certificate/:cert_name Produces: 200 application/json
+
+        :param cert_name: Name of the certificate
+        :type cert_name: str | unicode
+        :param mount_point: The "path" the aws auth method was mounted on
+        :return: The data key from the JSON response of the request.
+        :rtype: dict
+        """
+        api_path = '/v1/auth/{0}/config/certificate/{1}'.format(mount_point, cert_name)
+        response = self._adapter.get(
+            url=api_path,
+        )
+        return response.json().get('data')
```

Comment:
@jeffwecan we should consider standardizing json returns. personally i like this where the data key is returned directly, however, we have some that doesnt extract it. 

example: https://github.com/hvac/hvac/blob/c013371a43d30b5b54407f3fea2f2fd76bbe5934/hvac/api/auth_methods/ldap.py#L129